tests/hash/main.cpp

```diff
@@ -144,6 +152,55 @@ int main(int argc, char *argv[]) {
     }
   }
   input.open(argv[2], ios_base::in);
+
+  if(f == hash_blake2b) {
+      // blake2b does use different format and has key for hashing.
+      while(true)
+      {
+          char in[1024] = {0};
+          size_t inlen = 0;
+          // key_out is two buffer, first portion is key, second portion is out
+          char key_out[BLAKE2B_KEYBYTES + BLAKE2B_OUTBYTES] = {0};
+          char hash[64] = {0};
+          std::string t; // t as in temporary
+
+          input >> std::skipws >> t; // throw away if it is "in:" and break if nothing more is available to read
+          if (t.empty()) {
+            break;
+          } else if(test) { // first hash record special, does not have any "in:" value
+              input >> std::skipws >> t; // actual in data
+              t = boost::algorithm::unhex(t);
+              std::copy(t.begin(), t.end(), in);
+              inlen = t.size();
+          }
+          ++test;
+
+          input >> std::skipws >> t; // throw away "key:"
+          input >> std::skipws >> t; // actual keybytes data
+          t = boost::algorithm::unhex(t);
+          std::copy_n(t.begin(), BLAKE2B_KEYBYTES, key_out);
+
+          input >> std::skipws >> t; // throw away "hash:"
+          input >> std::skipws >> t; // actual hashbytes data
+          t = boost::algorithm::unhex(t);
+          std::copy(t.begin(), t.end(), hash);
+
+          f(in, inlen, key_out);
+
+          if(!std::equal(key_out + BLAKE2B_KEYBYTES, key_out + BLAKE2B_KEYBYTES + BLAKE2B_OUTBYTES, std::begin(hash))){
+              std::cerr << "For case number " << test
+                        << " computed hash value and given hash value does not match in blake2b";
+              return -1;
+          }
+
+          // Clean up the mess
+          memset(in, inlen, sizeof(char));
```

Comment:
Can you double check all 3 `memset`? The second and third parameter looks wrong.

Reply:
Nice catch. Although what we have is not required and is not incorrect. 

But semantically, it is incorrect. It only changes first byte. Will fix this.